contracts/DCAStrategies/DCAStrategies/DCAStrategiesPositionsHandler.sol

```diff
@@ -10,41 +10,48 @@ abstract contract DCAStrategiesPositionsHandler is IDCAStrategiesPositionsHandle
   // TODO: add function similar to this one https://github.com/Mean-Finance/dca-v2-core/blob/main/contracts/interfaces/IDCAHub.sol#L243
 
   /// @inheritdoc IDCAStrategiesPositionsHandler
-  function deposit(IDCAStrategies.DepositParams calldata parameters) external returns (uint256) {
-    IDCAStrategies.ShareOfToken[] memory _tokens = _getTokenShares(parameters.strategyId, parameters.version);
+  function deposit(IDCAStrategies.DepositParams calldata _parameters) external returns (uint256) {
+    IDCAStrategies.ShareOfToken[] memory _tokens = _getTokenShares(_parameters.strategyId, _parameters.version);
     if (_tokens.length == 0) revert InvalidStrategy();
 
-    IERC20(parameters.from).safeTransferFrom(msg.sender, address(this), parameters.amount);
+    IERC20(_parameters.from).safeTransferFrom(msg.sender, address(this), _parameters.amount);
 
+    uint256 _amountSpent;
     for (uint256 i = 0; i < _tokens.length; ) {
-      // uint256 _toDeposit = (parameters.amount * _tokens[i].share) / _TOTAL;
-      // _approveHub();
-      // IDCAPermissionManager.PermissionSet[] memory _permissions = new IDCAPermissionManager.PermissionSet[](0);
-      // parameters.hub.deposit(
-      //   parameters.from,
-      //   _tokens[i].token,
-      //   _toDeposit,
-      //   parameters.amountOfSwaps,
-      //   parameters.swapInterval,
-      //   address(this),
-      //   _permissions
-      // );
+      IDCAStrategies.ShareOfToken memory _token = _tokens[i];
+      uint256 _toDeposit = i < _tokens.length - 1 ? (_parameters.amount * _token.share) / _getTotalShares() : _parameters.amount - _amountSpent;
```

Comment:
Just realized that instead of calling `_getTotalShares` every time, we can store it in a memory variable before the loop. That should save some gas